library/swift/src/filters/FilterDataStatus.swift

```diff
@@ -35,4 +35,12 @@ public enum FilterDataStatus: Equatable {
   /// This may be called by filters which must parse a larger block of the incoming data before
   /// continuing processing, and will handle their own buffering.
   case stopIterationNoBuffer
+
+  /// Resume previously-stopped iteration, possibly forwarding headers if iteration was stopped
+  /// during an on*Headers invocation.
+  ///
+  /// It is an error to return `resumeIteration` if iteration is not currently stopped, and it is
+  /// an error to include headers if headers have already been forwarded to the next filter
+  /// (i.e. iteration was stopped during an on*Data invocation instead of on*Headers).
+  case resumeIteration(headers: T? = nil, data: Data)
```

Comment:
Something I noticed while testing out these interfaces on a filter migration is that it's not immediately obvious to the consumer that returning `.resumeIteration(...)` **does not call** `onResumeIteration()` - only calling `resumeRequest()` does. Thoughts on how we could clarify that either through naming or documentation?

Reply:
Would it make sense to simply rename `continue` and allow optional headers to be passed there? Since my understanding is the ability to pass headers here is the only difference between the two

Reply:
Based on our discussion elsewhere, I believe we've all settled on having "continue" be for continuing ongoing iteration and "resume" specifically for starting again when iteration has been stopped.